specs/validator/0_beacon-chain-validator.md

```diff
@@ -260,10 +260,15 @@ Let `get_eth1_data(block: Eth1Block) -> Eth1Data` be the function that returns t
 
 An honest block proposer sets `block.body.eth1_data = get_eth1_vote(state)` where:
 
+```python
+def compute_time_at_slot(state: BeaconState, slot: Slot) -> uint64:
+    return state.genesis_time + slot * SECONDS_PER_SLOT
+```
+
 ```python
 def voting_period_start_time(state: BeaconState) -> uint64:
-    eth1_voting_period_start_slot = state.slot % SLOTS_PER_ETH1_VOTING_PERIOD
-    return state.genesis_time + eth1_voting_period_start_slot * SECONDS_PER_SLOT
+    eth1_voting_period_start_slot = Slot(state.slot % SLOTS_PER_ETH1_VOTING_PERIOD)
```

Comment:
```suggestion
    eth1_voting_period_start_slot = state.slot - Slot(state.slot % SLOTS_PER_ETH1_VOTING_PERIOD)
```
It's pulled to alignment with `SLOTS_PER_ETH1_VOTING_PERIOD` right? Not from 0.

Reply:
nice catch. made it in the last PR.
Going to fix it up with using spec helpers

Reply:
nevermind. don't have good helpers for non-epoch based periods